core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -71,64 +72,129 @@ object ConsumerGroupCommand {
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        consumerGroupOutputWriter.printError("Executing consumer group command failed due to " + e.getMessage
+            + "\n" + Utils.stackTrace(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
+  case class ConsumerGroupAssignment(group: String, topicOpt: Option[String], partitionOpt: Option[Int], offsetOpt: Option[Long],
+                                     lagOpt: Option[Long], memberIdOpt: Option[String], memberHostOpt: Option[String],
+                                     clientIdOpt: Option[String], logEndOffset: Option[Long])
+
+  trait OutputWriter {
+    def printError(msg: String)
+    def printConfirmation(msg: String)
+    def printAssignmentHeader()
+    protected def printAssignmentRow(assignmentRow: ConsumerGroupAssignment)
+    def printAssignmentRows(assignmentRows: Array[ConsumerGroupAssignment]): Unit = {
+      assignmentRows.foreach { row => printAssignmentRow(row) }
+    }
+  }
+
+  class ConsumerGroupOutputWriter extends OutputWriter {
+
+    def printError(msg: String): Unit = println("Error: " + msg)
+
+    def printConfirmation(msg: String): Unit = println(msg)
+
+    def printAssignmentHeader(): Unit =
+      println("%-30s %-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format("GROUP", "TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "MEMBER-ID", "MEMBER-HOST", "CLIENT-ID"))
+
+    def printAssignmentRow(assignmentRow: ConsumerGroupAssignment): Unit =
+     println("%-30s %-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format(assignmentRow.group, assignmentRow.topicOpt.getOrElse(""),
+         assignmentRow.partitionOpt.getOrElse(""), assignmentRow.offsetOpt.getOrElse(""), assignmentRow.logEndOffset.getOrElse(""),
+         assignmentRow.lagOpt.getOrElse(""), assignmentRow.memberIdOpt.getOrElse("none"), assignmentRow.memberHostOpt.getOrElse(""),
+         assignmentRow.clientIdOpt.getOrElse("")))
+  }
+
   sealed trait ConsumerGroupService {
 
     def list(): Unit
 
-    def describe() {
-      describeGroup(opts.options.valueOf(opts.groupOpt))
+    def describe(): Option[Array[ConsumerGroupAssignment]] = {
+      describeGroup(opts.options.valueOf(opts.groupOpt)) match {
```

Comment:
nit: the name `describeGroup` no longer seems quite right. Maybe `assignmentState` would be more accurate since that's what the method returns.